actors/miner/tests/miner_actor_test_construction.rs

```diff
@@ -0,0 +1,163 @@
+use fil_actors_runtime::test_utils::*;
+use fil_actors_runtime::INIT_ACTOR_ADDR;
+
+use fil_actor_account::Method as AccountMethod;
+use fil_actor_miner::{
+    Actor, Deadline, Deadlines, Method, MinerConstructorParams as ConstructorParams, State,
+    WPOST_CHALLENGE_WINDOW, WPOST_PERIOD_DEADLINES,
+};
+
+use fvm_shared::address::Address;
+use fvm_shared::blockstore::CborStore;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::encoding::{blake2b_256, BytesDe, RawBytes};
+use fvm_shared::error::ExitCode;
+use fvm_shared::sector::{RegisteredPoStProof, SectorSize};
+
+use cid::Cid;
+
+mod util;
+
+#[allow(dead_code)]
+struct TestEnv {
+    receiver: Address,
+    owner: Address,
+    worker: Address,
+    worker_key: Address,
+    control_addrs: Vec<Address>,
+    peer_id: Vec<u8>,
+    multiaddrs: Vec<BytesDe>,
+    rt: MockRuntime,
+}
+
+fn prepare_env() -> TestEnv {
+    let mut env = TestEnv {
+        receiver: Address::new_id(1000),
+        owner: Address::new_id(100),
+        worker: Address::new_id(101),
+        worker_key: util::new_bls_addr(0),
+        control_addrs: vec![Address::new_id(999), Address::new_id(998)],
+        peer_id: vec![1, 2, 3],
+        multiaddrs: vec![BytesDe(vec![1, 2, 3])],
+        rt: MockRuntime::default(),
+    };
+
+    env.rt.receiver = env.receiver;
+    env.rt
+        .actor_code_cids
+        .insert(env.owner, *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.worker, *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.control_addrs[0], *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.control_addrs[1], *ACCOUNT_ACTOR_CODE_ID);
```

Comment:
You can replace this with something like:

```rust
env.rt.actor_code_cids
    .extend([env.owner, env.worker]
        .into_iter()
        .chain(env.control_addrs)
        .map(|addr|(*ACCOUNT_ACTOR_CODE_ID, addr)))
```

Reply:
yeah, but I am not sure it is better for the test.

the current way is braindead easy to read, the other way makes you think!!!

Reply:
I guess? It gets easier over time. But yeah, it's fine for the test.